cmux-tui/spec/remote-rpc.md

```
@@ -0,0 +1,302 @@
+| `workspace` | `id:string`, `root:string` |
+| `workspaces` | `workspaces:[[id,root]]` |
+| `stat` | `stat:FileStat` |
+| `file` | `data:ByteString`, `offset:u64`, `eof:bool`, `content_hash:string` |
+| `written` | `bytes:u64`, `content_hash:string` |
+| `directory` | `entries:[DirectoryEntry]`, `truncated:bool`, optional `next_cursor` |
+| `search` | `matches:[SearchMatch]`, `truncated:bool`, optional `next_cursor` |
+| `patch` | `changed_paths:[string]`, `applied:bool`, optional `files:[PatchFileResult]` default `[]` |
+| `git-status` | `status:{branch,head,changes}` |
+| `diff` | `data:ByteString`, `format:DiffFormat`, optional `next_cursor` |
+| `structured-diff` | `diff:StructuredDiffV1`, optional `next_cursor` |
+| `process-started` | `process:u64`, `pid:u32|null`, optional `operation:string` |
```

markdownlint-cli2 (0.23.0)

[warning] Spaces inside code span elements

(MD038, no-space-in-code)